### PR TITLE
Changes to email notification plugin

### PR DESCRIPTION
**Pull request by reshmabidikar**
- Fixed email templates (to make them in sync with aws templates)
- Updated unit tests for new email templates
- Disabled failing test due to Java versions
- Fixed issue in sending upcoming invoice email

### CODE REVIEW

**Comment by pierre** [comment]
Tests are now being run on GitHub Actions 🎉 but are failing on Java 8, most likely before of the whitespace changes. These tests must be commented out for now unfortunately (next year, Kill Bill 0.24.x will remove support for JDK 8, and we'll be able to re-enable them).

**Comment by reshmabidikar** [comment]
I've now disabled the failing tests and added a comment in the code.

File: src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java [hunk 1]
```diff
@@ -226,13 +226,15 @@ private void sendEmailForUpComingInvoice(final Account account, final ExtBusEven
         final DateTime targetDateTime = now.plus(span.getMillis());
 
         final PluginCallContext callContext = new PluginCallContext(EmailNotificationActivator.PLUGIN_NAME, now, context.getAccountId(), context.getTenantId());
+        final EmailNotificationConfiguration emailNotificationConfiguration = getConfiguration(context);
+        osgiKillbillAPI.getSecurityApi().login(emailNotificationConfiguration.getAdminUsername(), emailNotificationConfiguration.getAdminPassword());
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java:230]: Make sure to `logout` in a `finally` block, see https://github.com/killbill/killbill-qualpay-plugin/blob/90ae0b65ec57fc0f87d7718c4e120ab749d36c8e/src/main/java/org/killbill/billing/plugin/qualpay/QualpayPaymentPluginApi.java#L182-L188 for an example.

File: src/test/java/org/killbill/billing/plugin/notification/generator/TestTemplateRenderer.java [hunk 2]
```diff
@@ -126,129 +126,361 @@ public void beforeClass() throws Exception {
...
-                "If you have any questions about your account, please reply to this email or contact MERCHANT_NAME Support at: (888) 555-1234";
-
-        //System.err.println(email.getBody());
-        Assert.assertEquals(email.getSubject(), "Failed Payment");
-        Assert.assertEquals(email.getBody(), expectedBody);
-    }
+    
+    //Disabled as test fails on Java 8
+
+//    @Test(groups = "fast")
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/notification/generator/TestTemplateRenderer.java:132]: Could you actually disable them with `enabled = false` (like this: https://github.com/killbill/killbill/blob/00d9ace820bc7ddf6961a16a2727abe17b198bf9/invoice/src/test/java/org/killbill/billing/invoice/template/formatters/TestDefaultInvoiceItemFormatter.java#L60)? This is easier if we need to re-enable them manually for testing locally.

**Review by pierre — APPROVED** [review]
Thanks! I just have a couple of minor comments.

File: src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java (hunk 1 above)
Comment (reshmabidikar) [reply src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java]: Fixed

File: src/test/java/org/killbill/billing/plugin/notification/generator/TestTemplateRenderer.java (hunk 2 above)
Comment (reshmabidikar) [reply src/test/java/org/killbill/billing/plugin/notification/generator/TestTemplateRenderer.java]: Fixed

**Comment by pierre** [comment]
Thank you!

**Pull request merged**